Swiftfin/Views/UserDashboardView/ActiveSessionsView/ActiveSessionRowView.swift

```diff
@@ -0,0 +1,246 @@
+            L10n.playing.text
+                .foregroundColor(.green)
+                .font(.headline)
+        }
+    }
+
+    @ViewBuilder
+    private var playbackDetails: some View {
+        let displayPlayMethod = SessionPlaybackMethod.getDisplayPlayMethod(session)
+        HStack {
+            if displayPlayMethod != .unknown {
+                Text(displayPlayMethod.displayTitle)
+            }
+            Spacer()
+            playbackTimestamp
+        }
+        .font(.subheadline)
+        .foregroundColor(.secondary)
+    }
+
+    @ViewBuilder
+    private var playbackTimestamp: some View {
+        if let positionTicks = session.playState?.positionTicks,
+           let totalTicks = session.nowPlayingItem?.runTimeTicks
+        {
+            let elapsedTime = formattedTime(Int64(positionTicks))
+            let totalTime = formattedTime(Int64(totalTicks))
+            Text("\(elapsedTime) / \(totalTime)")
+        }
+    }
+
+    @ViewBuilder
+    private var playbackTimeline: some View {
+        if let progress = playbackProgress() {
+            ZStack(alignment: .leading) {
+                if let completionPercentage = session.transcodingInfo?.completionPercentage {
+                    ProgressView(value: completionPercentage / 100.0)
+                        .progressViewStyle(
+                            LinearProgressViewStyle(tint: .gray)
+                        )
+                        .scaleEffect(x: 1, y: 2)
+                }
+                ProgressView(value: progress)
+                    .progressViewStyle(
+                        LinearProgressViewStyle(tint: session.playState?.isPaused ?? false ? .orange : .green)
+                    )
+                    .scaleEffect(x: 1, y: 2)
+            }
+            .frame(height: 6)
+        }
+    }
+
+    private func formattedTime(_ ticks: Int64) -> String {
+        let seconds = ticks / 10_000_000
+        return seconds.timeLabel
+    }
+
+    private func playbackProgress() -> Double? {
+        guard let positionTicks = session.playState?.positionTicks,
+              let totalTicks = session.nowPlayingItem?.runTimeTicks,
+              totalTicks > 0
+        else {
+            return nil
+        }
+        return Double(positionTicks) / Double(totalTicks)
+    }
+
+    private func contentPoster(item: BaseItemDto, type: PosterDisplayType) -> AnyView {
+        var contentWidth: CGFloat
+        var imageSource: ImageSource?
+        var output: AnyView
+
+        switch type {
+        case .portrait:
+            contentWidth = 50
+            imageSource = item.portraitImageSources().randomElement()
+        case .landscape:
+            contentWidth = 133
+            imageSource = item.landscapeImageSources().randomElement()
+        }
+
+        if let cachedPoster = posterCache[item.id] {
+            output = cachedPoster
+        } else {
+            output = AnyView(
+                AsyncImage(url: imageSource?.url) { phase in
```

Comment:
I know this is a draft, but why remake `ImageView`?

Reply:
To be honest, I didn't see that exists! I'll move the images over to that.